src/plugins/intel_cpu/src/nodes/executors/shl/shl.hpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "shl_utils.hpp"
+#include "csinn/csinn_data_structure.h"
+#include "csinn/csinn_runtime.h"
+
+#include "memory_desc/cpu_memory_desc.h"
+
+#include <memory>
+
+
+namespace ov {
+namespace intel_cpu {
+
+
+template <typename T>
+struct ShlStructureTraits {};
+
+template <typename T, typename traits = ShlStructureTraits<T>>
+struct ShlStructure {
```

Comment:
Regarding the name and the placement of the class definition.
We might want to use this abstraction for other backends in the future as well.
But I guess for now it is fine to have it tied up with shl library.